packages/filebrowser-extension/src/index.ts

```diff
@@ -877,10 +880,16 @@ namespace Private {
         const { restored } = model;
 
         if (value.type === 'directory') {
-          return restored.then(() => model.cd(`/${localPath}`));
+          return restored.then(() => {
+            model.cd(`/${localPath}`);
```

Comment:
There is a subtle issue with this now, a change from the previous code. Previously, we returned a promise that resolved after the cd was accomplished. Now we are kicking off the cd and immediately returning this value. This would be easier to see if we rewrote the function using async/await syntax instead of the older Promise.then style code.

In async/await style, this previously was equivalent to:

```typescript
if (value.type === 'directory') {
  await restored;
  let result = await model.cd(...);
  return result
}
```
Now it is
```typescript
if (value.type === 'directory') {
  await restored;
  model.cd(...);
  return value
}
```
Notice the missing await? We aren't waiting for the model.cd to finish anymore.

I'd suggest rewriting the function using async/await to make it clearer exactly what is going on.

Reply:
> I'd suggest rewriting the function using async/await to make it clearer exactly what is going on.

(I'm happy to help you do this, perhaps early next week at the JLab sprint?)